include/zephyr/kernel.h

```diff
@@ -4820,18 +4820,19 @@ __syscall int k_msgq_put(struct k_msgq *msgq, const void *data, k_timeout_t time
  * pointer is not retained, so the message content will not be modified
  * by this function.
  *
+ * @note We consider put front an urgent action, so we assume the timeout parameter
+ * is always K_NO_WAIT.
```

Comment:
There, uh, is no timeout parameter. :)

I'd phrase this as "k_msgq_put_front() does not block".